test/core.jl

```diff
@@ -5089,3 +5089,34 @@ m22929_2.x = m22929_1
 @test !isdefined_22929_x(m22929_1)
 @test isdefined_22929_1(m22929_2)
 @test isdefined_22929_x(m22929_2)
+
+# issue 18933
+module GlobalDef18933
+    using Base.Test
+    # test that global declaration vs assignment operates correctly in local scope
+    f() = (global sin; nothing)
+    g() = (global cos; cos = 2; nothing)
+    @test @isdefined sin
+    @test !@isdefined cos
```

Comment:
It would be handy here to have some way to test what a binding refers to without dereferencing it, but that may be beyond the scope of this PR. E.g. `resolvebinding(:sin) == :(Base.sin)` and `resolvebinding(:cos) == :cos`.